users-mock/src/androidTest/java/pl/dybczak/githubusers/utils/TestBase.kt

```diff
@@ -0,0 +1,18 @@
+package pl.dybczak.githubusers.utils
+
+import android.app.Activity
+import org.junit.Rule
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+abstract class TestBase {
+
+    @RunWith(JUnit4::class)
+    abstract class TestBase<T : Activity>(activityClass: Class<T>) {
+
+        @JvmField
+        @Rule
+        val ruleChain: RuleChain = BaseTestRule.ruleChain(activityClass)
+    }
+}
```

Comment:
```suggestion
@RunWith(JUnit4::class)
abstract class TestBase<T : Activity>(activityClass: Class<T>) {

    @JvmField
    @Rule
    val ruleChain: RuleChain = BaseTestRule.ruleChain(activityClass)
}
```
Any reason to have TestBase nested inside TestBase?

Reply:
@piotrsulej  https://www.youtube.com/watch?v=QnMqRTu4Rcc